network/f5/bigip_selfip.py

```diff
@@ -0,0 +1,444 @@
+
+DOCUMENTATION = '''
+---
+module: bigip_selfip
+short_description: Manage Self-IPs on a BIG-IP system
+description:
+  - Manage Self-IPs on a BIG-IP system
+version_added: "2.2"
+options:
+  address:
+    description:
+      - The IP addresses for the new self IP. This value is ignored upon update
+        as addresses themselves cannot be changed after they are created.
+  name:
+    description:
+      - The self IP to create
+    required: false
+    default: Value of C(address)
+  netmask:
+    description:
+      - The netmasks for the self IP
+    required: true
+  state:
+    description:
+      - The state of the variable on the system. When C(present), guarantees
+        that the Self-IP exists with the provided attributes. When C(absent),
+        removes the Self-IP from the system.
+    required: false
+    default: present
+    choices:
+      - absent
+      - present
+  traffic_group:
+    description:
+      - The traffic group for the self IP addresses in an active-active,
+        redundant load balancer configuration
+    required: false
+  vlan:
+    description:
+      - The VLAN that the new self IPs will be on
+    required: true
+notes:
+  - Requires the f5-sdk Python package on the host. This is as easy as pip
+    install f5-sdk
+  - Requires the netaddr Python package on the host
+extends_documentation_fragment: f5
+requirements:
+  - netaddr
+  - f5-sdk
+author:
+  - Tim Rupp (@caphrim007)
+'''
+
+EXAMPLES = '''
+- name: Create Self IP
+  bigip_selfip:
+      address: "10.10.10.10"
+      name: "self1"
+      netmask: "255.255.255.0"
+      password: "secret"
+      server: "lb.mydomain.com"
+      user: "admin"
+      validate_certs: "no"
+      vlan: "vlan1"
+  delegate_to: localhost
+
+- name: Delete Self IP
+  bigip_selfip:
+      name: "self1"
+      password: "secret"
+      server: "lb.mydomain.com"
+      state: "absent"
+      user: "admin"
+      validate_certs: "no"
+  delegate_to: localhost
+'''
+
+RETURN = '''
+address:
+    description: The address for the Self IP
+    returned: created
+    type: string
+    sample: "192.168.10.10"
+name:
+    description: The name of the Self IP
+    returned:
+        - created
+        - changed
+        - deleted
+    type: string
+    sample: "self1"
+netmask:
+    description: The netmask of the Self IP
+    returned:
+        - changed
+        - created
+    type: string
+    sample: "255.255.255.0"
+traffic_group:
+    description: The traffic group that the Self IP is a member of
+    return:
+        - changed
+        - created
+    type: string
+    sample: "traffic-group-local-only"
+vlan:
+    description: The VLAN set on the Self IP
+    return:
+        - changed
+        - created
+    type: string
+    sample: "vlan1"
+'''
+
+try:
+    from f5.bigip import ManagementRoot
+    from icontrol.session import iControlUnexpectedHTTPError
+    HAS_F5SDK = True
+except ImportError:
+    HAS_F5SDK = False
+
+try:
+    from netaddr import *
```

Comment:
Pet peeve: blind import of * into the namespace. I personally feel `from netaddr import IPAddr` is safer and makes the code easier to read.